agent/xds/listeners.go

```diff
@@ -859,6 +859,10 @@ func (s *ResourceGenerator) makeInboundListener(cfgSnap *proxycfg.ConfigSnapshot
 		if err != nil {
 			return nil, err
 		}
+		if meshConfig := cfgSnap.MeshConfig(); meshConfig == nil || meshConfig.HTTP == nil || !meshConfig.HTTP.SanitizeXForwardedClientCert {
+			filterOpts.forwardClientDetails = true
+			filterOpts.forwardClientPolicy = envoy_http_v3.HttpConnectionManager_APPEND_FORWARD
```

Comment:
FYI @rboyer this is making it seem like with peering L7 applications could end up with two forwarded certificates. The one for the mesh gateway and the one for the source service. 

I don't think that's necessarily bad, but should probably be documented somewhere. Before peering there will only ever be one certificate present because gateways don't terminate mTLS.